core/src/main/java/org/wildfly/extras/creaper/core/online/SslOptions.java

```diff
@@ -0,0 +1,232 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+/**
+ * SSL/TLS connection options. Use {@code SslOptions.Builder} to create a new instance.
+ */
+public final class SslOptions {
+
+    private static final String DEFAULT_PROTOCOL = "TLS";
+    private static final String DEFAULT_STORE_TYPE = "JKS";
+
+    final String protocol;
+
+    final String keyStorePath;
+    final String keyStorePassword;
+    final String keyStoreType;
+    final String keyAlias;
+    final String keyPassword;
+
+    final String trustStorePath;
+    final String trustStorePassword;
+    final String trustStoreType;
+
+
+    private SslOptions(Builder builder) {
+        this.protocol = builder.protocol != null ? builder.protocol : DEFAULT_PROTOCOL;
+
+        this.keyStorePath = builder.keyStorePath;
+        this.keyStorePassword = builder.keyStorePassword;
+        this.keyStoreType = builder.keyStoreType != null ? builder.keyStoreType : DEFAULT_STORE_TYPE;
+        this.keyAlias = builder.keyAlias;
+        this.keyPassword = builder.keyPassword;
+
+        this.trustStorePath = builder.trustStorePath;
+        this.trustStorePassword = builder.trustStorePassword;
+        this.trustStoreType = builder.trustStoreType != null ? builder.trustStoreType : DEFAULT_STORE_TYPE;
+    }
+
+
+    public static final class Builder {
+        private String protocol;
+
+        private String keyStorePath;
+        private String keyStorePassword;
+        private String keyStoreType;
+        private String keyAlias;
+        private String keyPassword;
+
+        private String trustStorePath;
+        private String trustStorePassword;
+        private String trustStoreType;
+
+        /** Changes SSL/TLS protocol used for connection. Optional. TLS is used by default. */
+        public Builder protocol(String protocol) {
+            this.protocol = protocol;
+            return this;
+        }
+
+        /** Keystore with certificate for authentication. Optional. */
+        public Builder keyStore(String keyStorePath, String keyStorePassword) {
+            this.keyStorePath = keyStorePath;
+            this.keyStorePassword = keyStorePassword;
+            return this;
+        }
+
+        /** Key from keystore used for authentication. Optional. */
+        public Builder key(String keyAlias, String keyPassword) {
+            this.keyAlias = keyAlias;
+            this.keyPassword = keyPassword;
+            return this;
+        }
+
+        /** Changes expected type of keystore. Optional. JKS is expected by default. */
+        public Builder keyStoreType(String keyStoreType) {
```

Comment:
It could be an enum, I'll create one.

http://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#KeyStore